pandas/core/indexes/timedeltas.py

```diff
@@ -131,10 +132,18 @@ def __new__(cls, data=None, unit=None, freq=None, start=None, end=None,
                 periods=None, closed=None, dtype=None, copy=False,
                 name=None, verify_integrity=True):
 
+        if verify_integrity is not True:
```

Comment:
why isn't the default of verify_integrity None?

Reply:
right but aren't you deprecating it?

Reply:
Yes.  In the future it will just be set to `verify_integrity=True` at the top of `__new__`.  (and when the time comes, in the TDA/DTA constructors it will always be True.

Reply:
We still need the variable to exist because there are cases in which we can determine it is not necessary, in which case we can skip a potentially-expensive check.

Reply:
> We still need the variable to exist because there are cases in which we can determine it is not necessary, in which case we can skip a potentially-expensive check.

what is the expensive check?

Reply:
calling `inferred_freq = frequencies.infer_freq(self)` to check that this matches the user-supplied `freq`

Reply:
hmm, that's pretty annoying here. I think we talked about just deprecating ``freq`` in the constructor as well, yes? wouldn't this solve the problem? (or is this othorgonal).

could maybe just pass in a private arg instead ``_freq``? and deprecate verify_integrity?

Reply:
I don't recall that (and dont see how it would work), but yah, not passing freq would make verify_integrity unnecessary.

Reply:
> We still need the variable to exist because there are cases in which we can determine it is not necessary, in which case we can skip a potentially-expensive check.

@jbrockmendel Isn't the end goal to actually remove the `verify_integrity` keyword from the Index constructors? (not just keep it with a fixed True value). Because otherwise, why are you deprecating `verify_integrity=False` if we actually want to use it ourselves? 
In cases that we determine the check is not necessary, we can use `_simple_new` ?

Reply:
Eventually verify-integrity will not be in the signature.  The first line of the method will just define it to be True.

Reply:
done